pkg/helpers/tokencmd/basicauth.go

```diff
@@ -112,19 +141,29 @@ var basicRegexes = []*regexp.Regexp{
 	regexp.MustCompile(`(?i)^\s*basic(?:\s+|$)`),
 }
 
+// basicRealm returns true if a header indicates a basic auth challenge.
+// basicRealm returns the realm if one exists.  A non openshift:kubeadmin
+// basic challenge is favored.
 func basicRealm(headers http.Header) (bool, string) {
+	matchedBootstrap := false
 	for _, challengeHeader := range headers[http.CanonicalHeaderKey("WWW-Authenticate")] {
 		for _, r := range basicRegexes {
 			if matches := r.FindStringSubmatch(challengeHeader); matches != nil {
-				if len(matches) > 1 {
-					// We got a realm as well
-					return true, matches[1]
+				if !strings.Contains(challengeHeader, bootstrapRealm) {
+					if len(matches) > 1 {
+						return true, matches[1]
+					}
+					// No realm, but still basic
+					return true, ""
```

Comment:
@stlaz, should `basic no realm` be favored over `basic openshift:kubeadmin`?

Reply:
yes, it should